repo2docker/buildpacks/guix/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""BuildPack for guix environments"""
+import os
+
+from ..base import BuildPack, BaseImage
+
+
+class GuixBuildPack(BaseImage):
+    """A guix Package Manager BuildPack"""
+
+    def get_path(self):
+        """Return paths to be added to PATH environment variable"""
+        return super().get_path() + [
+            "/var/guix/profiles/per-user/$NB_USER/.guix-profile/bin"
+        ]
+
+
+    def get_build_scripts(self):
+        """
+        Install guix package manager version 1.3.0.x86_64-linux, using
+        an unmodified installation script found at
+        https://git.savannah.gnu.org/cgit/guix.git/plain/etc/guix-install.sh?id=a0178d34f582b50e9bdbb0403943129ae5b560ff
+        """
+        return super().get_build_scripts() + [
+            (
+                "root",
+                """
+                bash /tmp/.local/bin/guix-install.bash
+                """,
+            ),
+
+        ]
+
+    def get_build_script_files(self):
+
+        """Copying guix installation script on the image"""
+        return {
+            "guix/guix-install.bash":
+                "/tmp/.local/bin/guix-install.bash",
+        }
+
+    def get_assemble_scripts(self):
+        """
+        Wake up the guix daemon with root permission, set guix environnement
+        variables, make sure we never use debian's python by error by
+        renaming it, then, as an user install packages listed in
+        manifest.scm, use guix time-machine if channels.scm file exists
+        """
+        assemble_script ="""
+                 /var/guix/profiles/per-user/root/current-guix/bin/guix-daemon \
+                 --build-users-group=guixbuild --disable-chroot & \
+                 mv /usr/bin/python /usr/bin/python.debian && \
```

Comment:
I added some comments about this steps in the header of the function, let me now what you think.

  The point of using Guix is to avoid any confusions about the version or the provenance of the software you want to use and therefore guarantee and allow reproducibility of a software environment. Therefore having a python that we now nothing about could jeopardize that. Also counting on the path precedence to deal with that give us no guarantee of software environment reproducibility and could influence our results by making things work when they shouldn't for example.

While testing the Guix buildpack, I tried installing and using numpy, it take me a long time to realize that I where trying to do so but with the python already installed on the docker image, wich could not work with numpy (versions conflict). Also it can be confusing when the software use environment variables and you are not aware of all the other software present on the system.